jpa/subsystem/src/main/java/org/jboss/as/jpa/transaction/TransactionUtil.java

```diff
@@ -176,12 +179,19 @@ private void safeCloseEntityManager() {
             if ( afterCompletionCalled == true && associationCounter == 0) {
                 if (manager != null) {
                     try {
-                        if (ROOT_LOGGER.isDebugEnabled())
-                            ROOT_LOGGER.debugf("%s: closing entity managersession", getEntityManagerDetails(manager, scopedPuName));
-                        manager.close();
+                        if (cache != null) {
+                            if (ROOT_LOGGER.isDebugEnabled())
+                                ROOT_LOGGER.debugf("%s: putting transactional scoped entity manager back into cache", getEntityManagerDetails(manager, scopedPuName));
+                            manager.clear();
```

Comment:
We are calling clear to get the entity manager ready to be put in the cache, so that it can be reused.  The call to clear() ensures that no entities are associated with the entity manager.

The close() method does not update the database, since this entity manager was joined to a JTA transaction, the database update would of been done as part of the JTA transaction (e.g. some time before the Synchronization.afterCompletion() has been called, so either during Synchronization.beforeCompletion() or prior to that via application calling EntityManager.flush()).